In [2]:
from open_rcs import launch_rcs_widget

widget = launch_rcs_widget("..")

I'm doing some profiling and comparison and it seems that the precomputation optimization reduced the run time by 29%, a significant improvement, but we need more.  I want you to try numba (or if you have a better solution, we could do something else).  The multiprocessing is very interesting, but I'm concerned because multiprocessing doesn't work in jupyter kernels, for that we need to use ipyparallels which has a lot more overhead.

In [1]:
import os

os.chdir("../examples")

from profile_sphere_monostatic import run_profile

results = run_profile("..")

         5139586 function calls (4892490 primitive calls) in 3.964 seconds

   Ordered by: cumulative time
   List reduced from 3369 to 60 due to restriction <60>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.016    0.016    3.968    3.968 rcs_monostatic.py:21(simulate_monostatic)
     1369    0.006    0.000    3.900    0.003 rcs_functions.py:1700(accumulate_monostatic_sample_numba)
        1    0.000    0.000    3.700    3.700 dispatcher.py:344(_compile_for_args)
     17/1    0.000    0.000    3.688    3.688 dispatcher.py:862(compile)
  541/275    0.001    0.000    3.292    0.012 compiler_lock.py:32(_acquire_compile_lock)
      4/1    0.000    0.000    3.273    3.273 dispatcher.py:79(compile)
      4/1    0.000    0.000    3.273    3.273 dispatcher.py:86(_compile_cached)
      4/1    0.001    0.000    3.273    3.273 dispatcher.py:101(_compile_core)
      4/1    0.000    0.000    3.273    3.273 compiler.py:713(compile_extra)
      9/1    0.000 

In [3]:
1 - 3.8 / 915

0.9958469945355192

# Profiling History

## Inputs
- Model: F-35
- Surface Roughness and Correlation Distance: 0.0001m
- Theta Range: 6-120deg
- Phi and Theta Step: 3

Total: 42.68M evals

## Baseline
Total time: 915sec

## Optimization 1:
Total Time: 645sec

Run Time Reduction from Baseline: 29.5%

## Optimization 2:
Total Time: 297sec

Run Time Reduction from Baseline: 67.5%

## Optimization 3:
Total Time: 3.8sec

Run Time Reduction from Baseline: 99.6%